### DISCUSSION TITLE
Setting up Edlib on premise

### DISCUSSION THREAD

**Discussion by tmkarlsen** [cerpus/Edlib#1745] [General]
1. Start with following the [instructions for setup](https://docs.edlib.com/docs/developers/getting-started) documented on docs.edlib.com. Note: there are some known issues that we are working on to make this process complete and finish in one run.
2. The [Helm charts](https://github.com/cerpus/Edlib/tree/master/chart) describes how to set up Kubernetes (k8s) in the Edlib repo it self (this is basically Terraform, but for k8s).

We'll add more steps and improve the setup over time and as the process improves or changes.

Report back in this thread, and we'll do our best to help you out!

**Comment by torinfo** [comment]
Install Edlib3 on premise using LXD containers.

At the moment I am working on setting up Edlib (hub and content author) in two different LXD containers on Ubuntu 22.04 based on the Edlib3 branch.

See https://github.com/torinfo/Edlib/blob/master/INSTALL_LXD_CONTAINERS.md

The Hub is up and running, but I get an error when I try to launch the Content Author or the CA Admin modules. See attached screencast.

I added the tools using php artisan add-tool when the contant author was not yet up and running. Could that be the issue? It would be great if you have some pointers for me.
![edlib3_contantauthor](https://github.com/user-attachments/assets/ec70a8e0-9679-43c7-b83f-78f3f29d2392)

**Comment by torinfo** [comment]
In the mean time, I also tried to delete the records in the lti_tools (of the hub) and then to re-add them. That did not work.

**Comment by torinfo** [comment]
Also, updated from the edlib3 barnch to master. That process was succesful, but still the same error. I am missing soemting, but I don't know what.

Please also not that I chose to host the contentauthor in a subfolder of the edlib.digisprong.be domain. I don't know if that could be causing issues.

**Reply by emmachughes** [reply]
This will cause issues. CA was not made with hosting in a subdirectory in mind.

**Reply by torinfo** [reply]
Thank you for your very quick reply! I'll set it up on a sub-domain, so edlib.digisprong.be for the hub and ca.edlib.digisprong.be for the content author. Would that work?

**Reply by emmachughes** [reply]
Yep.

**Reply by torinfo** [reply]
I'll let you know, and I'll adapt the INSTALL instructions accordingly.

**Comment by torinfo** [comment]
I've updated to master, and realised that I needed PHP8.3 for the Content Author as well, so, installed PHP8.3.
Chaged the settings to run Content Author from ca.edlib.domain.com and now I get a lot further.
It seems to load the content author, and H5P view and Loading...

In the console I get ther following errors:

GET https://ca.edlib.domain.com/h5p-php-library/styles/h5p.css?ver=2.0.3 net::ERR_ABORTED 404 (Not Found)
GET https://ca.edlib.domain.com/h5p-php-library/styles/h5p-confirmation-dialog.css?ver=2.0.3 net::ERR_ABORTED 404 (Not Found)
GET https://ca.edlib.domain.com/h5p-php-library/styles/h5p-tooltip.css?ver=2.0.3 net::ERR_ABORTED 404 (Not Found)
GET https://ca.edlib.domain.com/h5p-php-library/styles/h5p-core-button.css?ver=2.0.3 net::ERR_ABORTED 404 (Not Found)
GET https://ca.edlib.domain.com/h5p-php-library/js/jquery.js?ver=2.0.3 net::ERR_ABORTED 404 (Not Found)
GET https://ca.edlib.domain.com/h5p-php-library/js/h5p-event-dispatcher.js?ver=2.0.3 net::ERR_ABORTED 404 (Not Found)
GET https://ca.edlib.domain.com/h5p-php-library/js/h5p.js?ver=2.0.3 net::ERR_ABORTED 404 (Not Found)
GET https://ca.edlib.domain.com/h5p-php-library/js/h5p-x-api-event.js?ver=2.0.3 net::ERR_ABORTED 404 (Not Found)
GET https://ca.edlib.domain.com/h5p-php-library/js/h5p-content-type.js?ver=2.0.3 net::ERR_ABORTED 404 (Not Found)
GET https://ca.edlib.domain.com/h5p-php-library/js/h5p-x-api.js?ver=2.0.3 net::ERR_ABORTED 404 (Not Found)
GET https://ca.edlib.domain.com/h5p-php-library/js/h5p-action-bar.js?ver=2.0.3 net::ERR_ABORTED 404 (Not Found)
GET https://ca.edlib.domain.com/h5p-php-library/js/h5p-confirmation-dialog.js?ver=2.0.3 net::ERR_ABORTED 404 (Not Found)
GET https://ca.edlib.domain.com/h5p-php-library/js/h5p-tooltip.js?ver=2.0.3 net::ERR_ABORTED 404 (Not Found)
GET https://ca.edlib.domain.com/h5p-php-library/js/request-queue.js?ver=2.0.3 net::ERR_ABORTED 404 (Not Found)
GET https://ca.edlib.domain.com/h5p-php-library/js/h5p-display-options.js?ver=2.0.3 net::ERR_ABORTED 404 (Not Found)
GET https://ca.edlib.domain.com/h5p-editor-php-library/scripts/h5peditor-editor.js?ver=2.0.3 net::ERR_ABORTED 404 (Not Found)
GET https://ca.edlib.domain.com/h5p-editor-php-library/scripts/h5peditor-init.js?ver=2.0.3 net::ERR_ABORTED 404 (Not Found)
GET https://ca.edlib.domain.com/h5p-editor-php-library/language/en.js?ver=2.0.3 net::ERR_ABORTED 404 (Not Found)

Any ideas?

**Reply by emmachughes** [reply]
These are symlinked from a directory in `vendor/`. You should be able to make the symlinks by running `php artisan storage:link`, otherwise you can run:

```
ln -fs ../vendor/h5p/h5p-core public/h5p-php-library
ln -fs ../vendor/h5p/h5p-editor public/h5p-editor-php-library
```

**Reply by torinfo** [reply]
@emmachughes Thanks again.
I suspected that, but as they have another name, I did not find them. I used php artisan storage:link. It creates **three** symbolic links:

ln -sf ../storage/app public/h5pstorage

And now it works!

![image](https://github.com/user-attachments/assets/bcd52acb-86ae-4552-bde3-14f970f1e631)

**Comment by torinfo** [comment]
In docker I see these folder in public folder of the edlib-contentauthor-1 container, but where do they come from?

**Reply by torinfo** [reply]
Should they have been installed by these commands:

php artisan h5p:library-hub-cache
php artisan h5p:library-install \\
     H5P.Accordion \\
     H5P.AppearIn \\
     H5P.Audio \\
     H5P.AudioRecorder \\
     H5P.Blanks \\
     H5P.Boardgame \\
     H5P.CoursePresentation \\
     H5P.Dialogcards \\
     H5P.DocumentationTool \\
     H5P.DragQuestion \\
     H5P.DragText \\
     H5P.Flashcards \\
     H5P.GreetingCard \\
     H5P.GuessTheAnswer \\
     H5P.IFrameEmbed \\
     H5P.ImageHotspotQuestion \\
     H5P.ImageHotspots \\
     H5P.ImpressPresentation \\
     H5P.InteractiveVideo \\
     H5P.MarkTheWords \\
     H5P.MemoryGame \\
     H5P.MultiChoice \\
     H5P.MultiMediaChoice \\
     H5P.QuestionSet \\
     H5P.Questionnaire \\
     H5P.SingleChoiceSet \\
     H5P.Summary \\
     H5P.Timeline \\
     H5P.TrueFalse \\
     H5P.TwitterUserFeed
